e2e/terraform/packer/ubuntu-bionic-amd64/dnsconfig.sh

```diff
@@ -1,18 +1,27 @@
 #!/bin/bash
-# Tasks in the Linux userdata can't be executed during AMI builds because they
-# rely on instance-specific data, but they only need to be executed once per
-# instance and not on every provisioning of Nomad
+set -e
+
+# These tasks can't be executed during AMI builds because they rely on
+# instance-specific data.
 
 # Add hostname to /etc/hosts
-echo "127.0.0.1 $(hostname)" | sudo tee --append /etc/hosts
+echo "127.0.0.1 $(hostname)" | tee --append /etc/hosts
+
+# this script should run after docker.service but we can't guarantee
+# it's created docker0 yet, so wait to make sure
+while :
+do
+    ip link | grep -q docker0 && break
```

Comment:
I may suggest rewriting as:

```suggestion
while ! (ip link | grep -q docker0)
do
```